codex/blockexchange/engine/engine.nim

```diff
@@ -571,13 +580,9 @@ proc taskHandler*(b: BlockExcEngine, task: BlockExcPeerCtx) {.gcsafe, async.} =
 
       codex_block_exchange_blocks_sent.inc(blocksDelivery.len.int64)
 
-      trace "About to remove entries from peerWants", blocks = blocksDelivery.len, items = task.peerWants.len
-      # Remove successfully sent blocks
-      task.peerWants.keepIf(
-        proc(e: WantListEntry): bool =
-          not blocksDelivery.anyIt( it.address == e.address )
-      )
-      trace "Removed entries from peerWants", items = task.peerWants.len
+      let deliveryAddresses = blocksDelivery.mapIt(it.address)
```

Comment:
`blocksDelivery.mapIt(it.address)` is already represented as `successAddresses`

Reply:
Glad you caught these. Good improvement.